staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -113,6 +113,24 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// BearerTokenSource for managing the token, if it's not nil, will use this to wrap the http request,
+	// if it's nil, and "AuthProvider" is not nil, we will generate one by "AuthProvider" configuration, so
+	// you can use this to manage and dynamic refresh the token.
+	BearerTokenSource BearerTokenSource
+}
+
+// BearerTokenSource for managing the token
+type BearerTokenSource interface {
+
+	// Token returns a bearer token for HTTP requests. Implementations may rotate the token
+	// returned by this method, for example, refreshing expired tokens.
+	Token() (string, error)
+
+	// Refresh is called when the REST client experiences a 401, which implies that the token is invalid
+	// or has expired. Implementations that do rotation automatically should return (false, nil) indicating
+	// the request should not be retried.
+	Refresh() (retry bool, err error)
```

Comment:
My initial though was that the Refresh() func should be always renew a token and cached it, interface should be more like this:
```
// Refresh will refresh a new token straightforward.
Refresh() (succeed bool, err error)
```
this will make sure the client side can control to rotate the token.

But current logic is that Token() will return the cached "valid" token, and Refresh() will return false( means do not need refresh), so  we cannot refresh a new token if the current token is valid. considering the scenario: the token is valid, not expired but authentication failed, we need refresh a new token.

@ericchiang WDYT ?